File: 2. Serving Patterns/2.2. Decorator Function/decorator.js
const http = require('http');

module.exports = {
    handler: (req, res) => {
        console.log(req.body);

        var obj = req.body;
        if (obj['Name'] === undefined) {
            obj['Name'] = 'Nameless';
        }
        if (obj['Color'] === undefined) {
            obj['Color'] = 'Transparent';
        }

        var options = {
            hostname: '52.165.18.22',
            port: 8080,
            path: '/',
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            }
        };

        var request = http.request(options, (respone) => {

            let data = '';

            respone.on('data', (chunk) => {
                data += chunk;
                console.log('data =' + data);
            });

            respone.on('end', () => {
                console.log('data =' + data);
                res.setHeader('Content-Type', 'application/json');
                res.end(data);
            });

        });

        request.on('error', (err) => {
            console.log('Error: ' + err.message);
            res.end('Error: ' + err.message);
        });

        request.write(JSON.stringify(obj));
        request.end();

    }
};

File: 2. Serving Patterns/2.2. Decorator Function/decorator.py
def handler(context):
  
  # We need the requests module to make HTTP requests to the Main function
  import requests
  # We need the json module to convert our JSON object to a JSON string
  import json

  # Make sure we create JSON object form our StringIO input
  print context.body.getvalue()
  obj = json.loads(context.body.getvalue())

  # Here we have some defaulting logic that will add the attributes if they do not exist
  if obj.get("Name", None) is None:
    obj["Name"] = "Nameless"
  if obj.get("Color", None) is None:
    obj["Color"] = "Transparent"

  # Make sure we perform our HTTP request to the Main function with a JSON content type header
  headers = {'content-type': 'application/json'}

  # Perform the actual call to our Main function
  # NOTE: Please change the IP address to the EXTERNAL-IP address from the main-storeshape-deployment (run this command: kubectl get services)
  r = requests.post(url = "http://52.165.129.49:8080", data=json.dumps(obj), headers=headers)
  
  print 'data =' + r.text

  return r.text